src/lucky_record/model.cr

```diff
@@ -1,16 +1,25 @@
 require "db"
 require "levenshtein"
+require "uuid"
 require "./schema_enforcer"
 
+
 class LuckyRecord::Model
   include LuckyRecord::Associations
   include LuckyRecord::SchemaEnforcer
 
   macro inherited
     FIELDS = [] of {name: Symbol, type: Object, nilable: Bool, autogenerated: Bool}
     ASSOCIATIONS = [] of {name: Symbol, foreign_key: Symbol, type: Object}
+  end
+
+  macro setup_autogenerated_columns(primary_key_type)
+    {% if primary_key_type == :uuid %}
+      column id : UUID, autogenerated: true
```

Comment:
@paulcsmith  The `UUID` here is what I want to make work. In contrast to `String` previously.

Reply:
Gotcha, and where is it still treating id as an `Int32`?

Reply:
If I say

```patch
  macro setup_initialize
    def initialize(
        {% for field in FIELDS %}
          @{{field[:name]}},
        {% end %}
      )
+      @id.foo
    end
  end
```

```
in macro 'setup_initialize' /Users/hanneskaeufler/Projects/forks/lucky/lucky_record/src/lucky_record/model.cr:62, line 18:

   1.     def initialize(
   2.
   3.           @id,
   4.
   5.           @created_at,
   6.
   7.           @updated_at,
   8.
   9.           @name,
  10.
  11.           @age,
  12.
  13.           @nickname,
  14.
  15.           @joined_at,
  16.
  17.       )
> 18.       @id.foo
  19.     end
  20.

undefined method 'foo' for Int32
```

Reply:
Oh I see. That's because the initializer accepts the raw column type. So in that case you'd do something like the spec here for CustomEmail: https://github.com/luckyframework/lucky_record/blob/ff9dde975d56dce436c77fe9bcef89a1c4ec4bad/spec/model_spec.cr#L84-L86

I *think* that will get you closer, but I'll try to take a closer look tonight :)

Reply:
It did get me closer 👍 , but I find that weird :) Why do I need to pass a String into the initializer if I have a UUID column?

Reply:
The DB.mapping passes those values to the initializer, so it takes the value the the DB.mapping uses. In this case, a String. 

It is a bit confusing, but in reality you should never use the initializer directly. It's only ever done in the specs, and even then, it is super rare

Reply:
Ahh, that explains it well, thanks